firmware-support/bittide-sys/src/axi.rs

```diff
@@ -85,23 +99,24 @@ impl AxiRx {
     pub fn try_receive(&self, buffer: &mut [u8]) -> Option<usize> {
         // Check if there is data to receive, this means either the buffer is full or a packet is complete
         // We can check by using the utility function read_status
-        if unsafe { self.status.read_volatile() == 0 } {
+        if self.read_status_raw() == 0 {
             return None;
         }
 
         // Get length of the incoming data
         let len = self.packet_length();
+
+        debug_assert!(len > buffer.len(), "Buffer too small to receive packet");
```

Comment:
Comparison should be the other way around here - if the boolean expression in `debug_assert!` evaluates to `false`, then a panic is triggered. So it should be `debug_assert!(len <= buffer.len(), "Buffer too small to receive packet");`.